Modified dashboard page

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->


#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Modified dashboard page UI to make it similar to the current EvalAI UI

<!-- Demo Link: Add here the ink where you changes can be seen. -->
- Link to live demo: http://pr-141-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
![Screenshot from 2019-06-06 00-30-19](https://user-images.githubusercontent.com/17106489/59008724-30645900-87f9-11e9-92b9-11e8eb59d1eb.png)

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Modified static header

Made header UI similar to Angularjs Application one.

sanketbansal committed:
Modified auth service

Added isAuth parameter in auth service

sanketbansal committed:
Modified test cases

Removed fake activated routes to work with routerLink  property in <a> tag

sanketbansal committed:
Modified test case of twitter feed component

Added mock window service for calling loadJS method.

sanketbansal committed:
Modified footer component

Added routerLink proprety for routing through UI

sanketbansal committed:
Modified footer component

Added github fork and star button using github api

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Modifications in header static component

Changes made in static header to handle different authentication cases

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Modified dashboard component

Changes made in dashboard component to make dashboard UI similar to angular JS application.

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Modified footer

Made the footer UI similar to the angularjs application one.

sanketbansal committed:
Added dashboard content component

Dashboard-content component added to added functionality in the dashboard UI

sanketbansal committed:
Added simple header component

simple header added to modify the UI of the header in the dashboard

sanketbansal committed:
Added side bar component

Added side bar as utility component for the side bar in the Dashboard UI

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

sanketbansal committed:
Fixed testing issue

Removed script to load github stars and forks.
Added routes for router testing module in dashboard content component

sanketbansal committed:
Moved dashboard component styles

Styles related to dashboard component moved from 'styles-port/modules/dashboard.scss'

sanketbansal committed:
Fixed lint error in angular.json file

Added comma after a property in angular.json file

sanketbansal committed:
Added materialize css lib

Materialize css lib added for animations and custom styles

sanketbansal committed:
Changes made in sidebar component

Corrected routerlink for participant teams

sanketbansal committed:
Removed images-port and styles-port folder

Styles and images assets removed from ported folder

sanketbansal committed:
Modified basic styles

Basic styles has been modifed to remove duplicate styles

sanketbansal committed:
Removed duplicate styles

Removed duplicate scss files from basic.scss

sanketbansal committed:
Removed unused css from main.scss

Duplicates removed from main.scss file

sanketbansal committed:
Removed console.logs and comments

Console logs and comments removed

sanketbansal committed:
Modified styles

Added styles into the specific component

sanketbansal committed:
Changes in auth service

Removed unecessary methods and variables

sanketbansal committed:
Removed comments

comment styles removed from dashboard-content component file

sanketbansal commented:
@lunayach this can be reviewed now

sanketbansal committed:
Fixed UI bug

overflow of  logo innav fixed in simple header component

lunayach commented on angular.json:
```diff
@@ -29,10 +29,12 @@
               "./node_modules/froala-editor/css/froala_style.min.css",
-              "./node_modules/ng-pick-datetime/assets/style/picker.min.css"
+              "./node_modules/ng-pick-datetime/assets/style/picker.min.css",
+              "src/styles/styles-port/basic.scss"
```
Didn't we agree on removing `styles-port`?

sanketbansal replied:
I am making a PR for that after merging that It will be removed. I kept it to make the PR functional.

lunayach commented on src/app/components/dashboard/dashboard-content/dashboard-content.component.ts:
```diff
@@ -0,0 +1,104 @@
+   */
+  ngOnInit() {
+    if (!this.authService.isLoggedIn()) {
+      this.router.navigate(['/auth/login']);
```
Don't hard-code the URL, `/auth/login`, define a separate variable above or get from elsewhere if available.

lunayach commented on src/app/components/dashboard/dashboard-content/dashboard-content.component.ts:
```diff
@@ -0,0 +1,104 @@
+  /**
+   * Participant teams list
+   */
+  participantteams = [];
```
Camel casing in variable names.

sanketbansal replied:
Done

lunayach commented on src/app/components/dashboard/dashboard.component.ts:
```diff
@@ -53,55 +43,17 @@ export class DashboardComponent implements OnInit {
-    );
+    this.authServiceSubscription = this.authService.change.subscribe((authState) => {
+      if (!authState.isLoggedIn) {
+        this.router.navigate(['/auth/login']);
```
Same here for `/auth/login`.

lunayach commented on src/app/components/nav/dash-footer/dash-footer.component.html:
```diff
@@ -0,0 +1,37 @@
+<footer class="ev-md-container dashboard-footer ev-dark-bg">
```
I had reviewed the same file in #158!
Should we close/merge that first?

sanketbansal replied:
I think we should close that PR

lunayach commented on src/app/components/nav/header-static/header-static.component.ts:
```diff
@@ -165,4 +177,6 @@ export class HeaderStaticComponent implements OnInit, OnDestroy {
   }
 
+  profileDropdown(event) {}
+
```
This function does nothing.

sanketbansal replied:
ok i will remove that.

Shekharrajak commented on src/app/services/auth.service.ts:
```diff
@@ -12,12 +12,21 @@ export class AuthService {
 
   /**
-   * Modifications in Auth Services
+   * Ported From Angular Application
```
Means ?

sanketbansal replied:
Okay, I will remove these comments.

sanketbansal replied:
done

Shekharrajak commented on src/app/services/auth.service.ts:
```diff
@@ -135,22 +144,24 @@ export class AuthService {
+          success();
         },
         err => {
+          error();
```
When this function will be used?

sanketbansal replied:
removed changes in it

Shekharrajak commented on src/app/components/nav/simple-header/simple-header.component.ts:
```diff
@@ -0,0 +1,149 @@
+  templateUrl: './simple-header.component.html',
+  styleUrls: ['./simple-header.component.scss']
+})
+export class SimpleHeaderComponent implements OnInit, OnDestroy {
```
Why its name is simple header ?

sanketbansal replied:
I named it as per used in angularJS project.

Shekharrajak commented on src/app/components/nav/simple-header/simple-header.component.ts:
```diff
@@ -0,0 +1,149 @@
+})
+export class SimpleHeaderComponent implements OnInit, OnDestroy {
+
+  user = {username: ''};
```
Just want to let you know - If you want to create an object then it would be better to user Angular interface concept and make a model in a separate file.

sanketbansal replied:
Okay, I will keep that in mind.

sanketbansal committed:
Added style for footer

Style added for footer components

sanketbansal committed:
Changes in auth services

Remove uunused changes from verify email method
Removed unused comments

sanketbansal committed:
Removed unused method

Removed profiledropdown method from static header

sanketbansal committed:
changes in dashboard component

Added routepath variable.
Added camel casing for variable.

sanketbansal committed:
Fixed bug in challenge submit component

Added padding ul element to overwrite default styles applied by materialize.scss

lunayach commented:
@sanketbansal resolve the conflicts.
Since, #182 and #158 have been merged, also clean this PR.

sanketbansal commented:
> @sanketbansal resolve the conflicts.
> Since, #182 and #158 have been merged, also clean this PR.

okay

sanketbansal committed:
Configuration changes

Removed styles-port folder
Removed duplicates from app module
removed styles-port/basic.sccs from angular.json

sanketbansal committed:
changes in base.scss file

Added proprtiess in .btn class

sanketbansal committed:
Removed unnecessary changes

Unnecessary changes removed from auth service

lunayach commented:
@sanketbansal please verify the changes once before pushing. This big image comes when visiting any link in the application:

![two](https://user-images.githubusercontent.com/21227893/61588085-c4328e80-abc7-11e9-89c5-f57e31626499.png)

sanketbansal commented:
@lunayach it's working fine on my machine.

![Screenshot from 2019-07-21 03-02-01](https://user-images.githubusercontent.com/17106489/61588121-f9a59080-ab63-11e9-8a18-06bffe78afa9.png)

lunayach commented on src/app/components/nav/footer/footer.component.html:
```diff
@@ -1,61 +1,37 @@
-	</div>
-</div>
-</div>
+<footer class="ev-md-container ev-footer ev-dark-bg">
```
@sanketbansal no need of this `footer` component again. It has the identical code for `dash-footer` component which is already merged in #158.

sanketbansal replied:
okay, will remove that.

sanketbansal replied:
should I rename the dash-footer to footer ?

lunayach replied:
Yes, please.

lunayach commented on src/app/components/nav/header-static/header-static.component.html:
```diff
@@ -1,35 +1,46 @@
+          <div class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-14" (click)="logOut()">Logout</div>
+        </li>
+      </ul>
+      <ul class="side-nav" style="width:200px; transform:translateX(0px);" (click)="menuExpander()" *ngIf="!isMenuExpanded" >
```
Move these inline styles to `.scss`.

sanketbansal replied:
okay

lunayach commented on src/app/components/nav/header-static/header-static.component.html:
```diff
@@ -1,35 +1,46 @@
+      </ul>
+      <ul class="right hide-on-med-and-down">
+        <li *ngIf="!authService.isAuth"><a class="waves-effect waves-dark main-header-link" routerLink="/auth/signup" routerLinkActive="active">Sign Up</a></li>
+        <li *ngIf="!authService.isAuth"><a class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-14" style="padding: 0 29px; font-size: 14px; display: inline-block" routerLink="/auth/login" routerLinkActive="active">Log In</a></li>
```
Here, as well, move `adding: 0 29px; font-size: 14px; display: inline-block` to `.scss`.

sanketbansal replied:
okay

lunayach commented on src/app/components/nav/header-static/header-static.component.ts:
```diff
@@ -102,18 +110,22 @@ export class HeaderStaticComponent implements OnInit, OnDestroy {
     this.innerWidth = window.innerWidth;
-    if (this.innerWidth <= 810) {
-      this.isMenuExpanded = false;
+    if (this.innerWidth > 810) {
```
Create an `expandMenu( )` method to use it in both the places i.e. Line 113 and Line 127 respectively.

sanketbansal replied:
okay

lunayach commented on src/app/components/nav/simple-header/simple-header.component.html:
```diff
@@ -0,0 +1,43 @@
+          <a class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-14" routerLink="/auth/login">Log In</a>
+        </li>
+      </ul>
+      <ul class="side-nav" style="width:200px; transform:translateX(0px);" (click)="menuExpander()" *ngIf="!isMenuExpanded">
```
Move `style="width:200px; transform:translateX(0px);"` to `.scss`. (Also, at other places.)

sanketbansal replied:
> > @lunayach it's working fine on my machine.
> > ![Screenshot from 2019-07-21 03-02-01](https://user-images.githubusercontent.com/17106489/61588121-f9a59080-ab63-11e9-8a18-06bffe78afa9.png)
> 
> Yes, re-building it helped. But as you might see, there is something wrong in the Home-page UI. Would this get fixed in #140?

yes for sure.

lunayach commented on src/app/components/nav/simple-header/simple-header.component.html:
```diff
@@ -0,0 +1,43 @@
+<div class="sim-header padding-lr-header">
```
**Simple Header** and **Header Static** component are very similar. Can you tell why do we need both of them?

sanketbansal replied:
okay, I will try to make a single component for both of them.

lunayach replied:
Thanks! That'll be great.

lunayach commented on src/app/components/nav/simple-header/simple-header.component.ts:
```diff
@@ -0,0 +1,149 @@
+  /**
+   * Header is transparent on these URLs
+   */
+  transparentHeaderUrls = ['', '/'];
```
I don't think, this is used anywhere.

sanketbansal replied:
okay, I will check and remove that

lunayach commented:
> @lunayach it's working fine on my machine.
> 
> ![Screenshot from 2019-07-21 03-02-01](https://user-images.githubusercontent.com/17106489/61588121-f9a59080-ab63-11e9-8a18-06bffe78afa9.png)

Yes, re-building it helped. But as you might see, there is something wrong in the Home-page UI. Would this get fixed in #140?

sanketbansal committed:
Changes made in static-header

UI bugs fixed in static header.

sanketbansal committed:
Removed simple header

Simple header component has been removed

sanketbansal committed:
Changes in static header

Added checkInnerwidth method

sanketbansal committed:
Changes in footer component

Added styles of dash footer component

sanketbansal committed:
Removed dash footer component

Dash footer has been removed as it has duplicate styles and html with footer component

lunayach commented:
@sanketbansal when I click on the available options on the side-bar (i.e. **All Challenges**, **Hosted Challenges**, etc.), the corresponding side-bar having these options dis-appears. (unlike current version). Can you please check this?

sanketbansal commented:
> @sanketbansal when I click on the available options on the side-bar (i.e. **All Challenges**, **Hosted Challenges**, etc.), the corresponding side-bar having these options dis-appears. (unlike current version). Can you please check this?

That's not a bug actually after clicking on the options it gets routed to other components which are 
being modified in different PR. There side bar remains intact.

lunayach commented:
@sanketbansal  The stats for the GitHub are not displayed in one go and I have to do a refresh and only then they are displayed.

sanketbansal committed:
Changes in footer component

Added new github forks and star button

lunayach reviewed (approved):
@RishabhJain2018 Now this can be merged.
It does break some UI of the homepage but that would be taken care with #140.

lunayach committed:
Merge branch 'master' into modified-dashboard
